pandas/core/frame.py

```diff
@@ -1978,7 +1978,48 @@ def memory_usage(self, index=True, deep=False):
         return result
 
     def transpose(self, *args, **kwargs):
-        """Transpose index and columns"""
+        """
+        Transpose index and columns.
+
+        Reflect the dataframe over its main diagonal by writing rows as columns
+        and vice-versa.
+
+        Parameters
+        ----------
+        *args
+            These arguments will be passed to
+            :func:`pandas.compat.numpy.function.validate_transpose`.
+        **kwargs
+            Keyword arguments used to create the transposed DataFrame.
+
+        Returns
+        -------
+        DataFrame
+            The transposed DataFrame.
+
+        See Also
+        --------
+        numpy.transpose : Permute the dimensions of a given array.
+
```

Comment:
Another recommendation to include in the *See Also* section:
DataFrame.T: alias for DataFrame.transpose

Reply:
That's a good idea, the only problem with that is that we use the same docstring for `.T`, so then it would be pointing to itself. 

So I would just reflect the full docstring for both, like you already did in the examples.

@igorcadelima maybe add in the extended summary that it is both as method transpose() and property T